### ISSUE TITLE
RGCN demo results are typically much lower than checked in version

### ISSUE THREAD

**Issue by huonw** [stellargraph/stellargraph#677]
**Describe the bug**
The RGCN demo notebook currently mentions an validation accuracy of 0.94:

https://github.com/stellargraph/stellargraph/blob/7689c1d16955db811085cc6ab9b1e6dc8cf84b98/demos/node-classification/rgcn/rgcn-aifb-node-classification-example.ipynb#L292-L294

However that does not seem easily reproducible, with validation accuracies closer to 0.4-0.6 on `develop` and even in the commit that landed the RGCN code:

| code | acc 0 | acc 1 | acc 2 | acc 3 |
|---|--:|--:|---:|---:|
|a2a2ef28|0.5|0.58|0.63|0.66|
|develop (13611b46651c453ceda48eb2e1b090e9f12b3da5)|0.36|0.38|0.47|0.53|

(Sorted validation accuracies as reported by 4 runs of the notebook on the relevant commit)

This could be related to #649.

**To Reproduce**
1. Open `demos/node-classification/rgcn/rgcn-aifb-node-classification-example.ipynb`
2. Run it
4. Observe the lower accuracy than the notebook in `develop`

**Expected behavior**
The checked-in notebook should be reproducible (or at least, a close result).

**Screenshots**
Training history with manual run in a2a2ef28:

![image](https://user-images.githubusercontent.com/1203825/72393134-1ecb3280-3786-11ea-9e5d-502a50962c50.png)

Training history of checked in notebook:

![image](https://user-images.githubusercontent.com/1203825/72393156-2e4a7b80-3786-11ea-9286-63a6fb0cf26b.png)

**Desktop (please complete the following information):**
 - OS: macOS

**Data used**
N/A

**Additional context**
N/A

**Comment by kjun9** [comment]
Also encountered while doing https://github.com/stellargraph/stellargraph/pull/669 - getting tensorflow warnings during each epoch of training:
```
WARNING:tensorflow:Gradients do not exist for variables ['relational_graph_convolution_1/bases:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_1/coeff:0', 'relational_graph_convolution_3/bases:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0', 'relational_graph_convolution_3/coeff:0'] when minimizing the loss.
1/1 [==============================] - 20s 20s/step - loss: 1.6059 - acc: 0.3944 - val_loss: 1.5863 - val_acc: 0.3611
```
![image.png](https://images.zenhubusercontent.com/5ad3f71c4b5806bc2bcf28ff/ada9aa2f-1541-4061-b4fe-d48e5b3e72d9)

**Comment by huonw** [comment]
> Also encountered while doing #669 - getting tensorflow warnings during each epoch of training:

That's #649.

**Comment by kieranricardo** [comment]
This issue was fixed with #731

**Comment by huonw** [comment]
I think we generally don't close issues directly, but instead move them to Done in Zenhub.
